### ISSUE TITLE
`pilot list` doesn't show Devices, Latest Version, or Latest Install Date

### ISSUE THREAD

**Issue by rmccamish** [fastlane/fastlane#4216]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

$ pilot list -p XXXXXXXX  

XXXXXX is our apple-id.    

First and last name and email are correct.  However, Devices is always 0 and Latest Versions and Install Date are blank.

##### Complete output when running fastlane, including the stack trace and command used

```
INFO [2016-04-14 16:05:33.39]: Login to iTunes Connect (xxxxxxx)
INFO [2016-04-14 16:05:36.85]: Login successful
+-------+----------+---------------------------+---------+----------------+---------------------+
|                                       Internal Testers                                        |
+-------+----------+---------------------------+---------+----------------+---------------------+
| First | Last     | Email                     | Devices | Latest Version | Latest Install Date |
+-------+----------+---------------------------+---------+----------------+---------------------+
| xxxx  | Smith    | xxxxxxxxxxxxx        | 0       |                |                     |

```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
# INSERT CONTENT HERE
```

### Environment

fastlane version (run `fastlane -v`):     fastlane 1.81.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? no

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?   no

**Comment by vpolouchkine** [comment]
Thanks for the report, we'll take a look.

**Comment by senfi** [comment]
I am observing the same with the same configuration/environment.

**Comment by KrauseFx** [comment]
Thanks for reporting this issue, taking a look at it now 👍

**Comment by TKBurner** [comment]
@rmccamish Thanks for sharing this report. We've seen another report like this and are going to look into it. Did you create the testers using Pilot? Thanks!

**Comment by rmccamish** [comment]
No the users already existed.

Thx!

Ron....

On Apr 26, 2016, at 1:14 PM, Todd Burner <notifications@github.com<mailto:notifications@github.com>> wrote:


@rmccamish<https://github.com/rmccamish> Thanks for sharing this report. We've seen another report like this and are going to look into it. Did you create the testers using Pilot? Thanks!

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/fastlane/fastlane/issues/4216#issuecomment-214815232>

*** *** ***
This message contains information which may be confidential and privileged. Unless you are the addressee (or authorized to receive for the addressee), you may not use, copy or disclose to anyone the message or any information contained in the message. If you have received the message in error,  please advise the sender by reply e-mail and delete the message.

**Comment by mfurtak** [comment]
I have reproduced this locally, and will look into it. Thanks! 👍

**Comment by mfurtak** [comment]
Hi folks! 👋 

The issue here is that Apple returns different data depending on how you ask for tester information. Counter-intuitively, when asking for a list of testers outside of the context of an app, you will find that the last three columns are filled with data.

(from someplace outside of your app directory)
```
pilot list -p your@apple.id
```

```
+----------+---------------+----------------------------------+---------+----------------+---------------------+
|                                               Internal Testers                                               |
+----------+---------------+----------------------------------+---------+----------------+---------------------+
| First    | Last          | Email                            | Devices | Latest Version | Latest Install Date |
+----------+---------------+----------------------------------+---------+----------------+---------------------+
| Whoever  | Someguy       | whoever.someguy@gmail.com        | 0       |                |                     |
| Another  | Person        | another.person@gmail.com         | 1       | 1.0 (21)       | 04/27/16 09:44      |
+----------+---------------+----------------------------------+---------+----------------+---------------------+

+----------+-----------+---------------------------+---------+----------------+---------------------+
|                                         External Testers                                          |
+----------+-----------+---------------------------+---------+----------------+---------------------+
| First    | Last      | Email                     | Devices | Latest Version | Latest Install Date |
+----------+-----------+---------------------------+---------+----------------+---------------------+
| Another  | Person    | another.person@gmail.com  | 1       | 1.0 (21)       | 04/27/16 09:44      |
| John     | Doer      | mail@domain.com           | 0       |                |                     |
+----------+-----------+---------------------------+---------+----------------+---------------------+
```

The version and install data is actually relative to a particular app, but we're not describing which one. 🙁

Then, when we ask for tester information in the context of an app, the shape of the data is quite different, so these fields are not available.

This is definitely a bug, or at least a feature design problem. But it will take some work to address properly. In the mean time, I'm inclined to remove those columns when we have no actual data available to display.

Does that make sense?

**Comment by rmccamish** [comment]
Makes sense to me - thanks for looking into it.   I'll run pilot outside of the context of an app and filter the data.

Again, thanks!

Ron....

On Apr 27, 2016, at 11:41 AM, Michael Furtak <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi folks! 👋

The issue here is that Apple returns different data depending on how you ask for tester information. Counter-intuitively, when asking for a list of testers outside of the context of an app, you will find that the last three columns are filled with data.

(from someplace outside of your app directory)

pilot list -p your@apple.id<mailto:your@apple.id>


+----------+---------------+----------------------------------+---------+----------------+---------------------+
|                                               Internal Testers                                               |
+----------+---------------+----------------------------------+---------+----------------+---------------------+
| First    | Last          | Email                            | Devices | Latest Version | Latest Install Date |
+----------+---------------+----------------------------------+---------+----------------+---------------------+
| Whoever  | Someguy       | whoever.someguy@gmail.com<mailto:whoever.someguy@gmail.com>        | 0       |                |                     |
| Another  | Person        | another.person@gmail.com<mailto:another.person@gmail.com>         | 1       | 1.0 (21)       | 04/27/16 09:44      |
+----------+---------------+----------------------------------+---------+----------------+---------------------+

+----------+-----------+---------------------------+---------+----------------+---------------------+
|                                         External Testers                                          |
+----------+-----------+---------------------------+---------+----------------+---------------------+
| First    | Last      | Email                     | Devices | Latest Version | Latest Install Date |
+----------+-----------+---------------------------+---------+----------------+---------------------+
| Another  | Person    | another.person@gmail.com<mailto:another.person@gmail.com>  | 1       | 1.0 (21)       | 04/27/16 09:44      |
| John     | Doer      | mail@domain.com<mailto:mail@domain.com>           | 0       |                |                     |
+----------+-----------+---------------------------+---------+----------------+---------------------+


The version and install data is actually relative to a particular app, but we're not describing which one. 🙁

Then, when we ask for tester information in the context of an app, the shape of the data is quite different, so these fields are not available.

This is definitely a bug, or at least a feature design problem. But it will take some work to address properly. In the mean time, I'm inclined to remove those columns when we have no actual data available to display.

Does that make sense?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/fastlane/fastlane/issues/4216#issuecomment-215124827>

*** *** ***
This message contains information which may be confidential and privileged. Unless you are the addressee (or authorized to receive for the addressee), you may not use, copy or disclose to anyone the message or any information contained in the message. If you have received the message in error,  please advise the sender by reply e-mail and delete the message.

**Comment by TKBurner** [comment]
Awesome @rmccamish! I'm going to go ahead and close this issue out 👍

**Comment by mfurtak** [comment]
I've merged changes in #4401 to make the list output better reflect the data we have available in each circumstance. This will go out in the next `pilot` release 🚀

**Comment by mfurtak** [comment]
These changes have shipped as part of [`pilot` 1.5.1](https://github.com/fastlane/fastlane/releases/tag/pilot%2F1.5.1)! ✈️
